File: CSharp/Yow.CoD.Finance.Domain/Services/DisburseLoanFundsCommandHandler.cs
using System.Threading.Tasks;
using Yow.CoD.Finance.Domain.Contracts;
using Yow.CoD.Finance.Domain.Model;

namespace Yow.CoD.Finance.Domain.Services
{
    public class DisburseLoanFundsCommandHandler : IHandler<DisburseLoanFundsCommand, Receipt>
    {
        private readonly IRepository<Loan> _repository;

        public DisburseLoanFundsCommandHandler(IRepository<Loan> repository)
        {
            _repository = repository;
        }

        public async Task<Receipt> Handle(DisburseLoanFundsCommand command)
        {
            var loan = await _repository.Get(command.AggregateId);
            loan.DisburseFunds(command);
            await _repository.Save(loan);
            return new Receipt(loan.Id, loan.Version);
        }
    }
}

File: typescript/packages/domain/src/services/disburse-loan-funds-command-handler.ts
import { DisburseLoanFundsCommand } from "../contracts/disburse-loan-funds-command.js";
import { Receipt } from "../contracts/receipt.js";
import { Loan } from "../model/loan.js";
import { Handler } from "./handler.js";
import { Repository } from "./repository.js";

export class DisburseLoanFundsCommandHandler
  implements Handler<DisburseLoanFundsCommand, Receipt>
{
  readonly #repository: Repository<Loan>;

  constructor(repository: Repository<Loan>) {
    this.#repository = repository;
  }

  async handle(command: DisburseLoanFundsCommand): Promise<Receipt> {
    const loan = await this.#repository.get(command.aggregateId);
    loan.disburseFunds(command);
    await this.#repository.save(loan);
    return new Receipt(loan.id, loan.version);
  }
}
